File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/core/implementation/SurveysServiceImpl.java
package io.zenwave360.example.clinicaltool.modules.surveys.core.implementation;

import io.zenwave360.example.clinicaltool.modules.surveys.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.inbound.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.inbound.dtos.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.implementation.mappers.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.outbound.jpa.*;

import java.math.*;
import java.time.*;
import java.util.*;
import org.mapstruct.factory.Mappers;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Service Implementation for managing [SurveyAnswers].
 */
@Service
@Transactional(readOnly = true)
@lombok.AllArgsConstructor
public class SurveysServiceImpl implements SurveysService {

    private final Logger log = LoggerFactory.getLogger(getClass());

    private final SurveysServiceMapper surveysServiceMapper = SurveysServiceMapper.INSTANCE;


    private final SurveyAnswersRepository surveyAnswersRepository;






    public SurveyAndQuestions getSurveyAndQuestionsForPatient(String name, Long patientId, String lang)
 {
    log.debug("Request getSurveyAndQuestionsForPatient: {} {} {}", name, patientId, lang);

    var surveyAnswers = surveysServiceMapper.update(new SurveyAnswers(), name, patientId, lang);
    // TODO: implement this method
    return surveysServiceMapper.asSurveyAndQuestions(surveyAnswers);


}


    public SurveyAnswers answerSurvey(Long surveyId, Long patientId, LocalDate date, SurveyAnswers input)
 {
    log.debug("Request answerSurvey: {} {} {} {}", surveyId, patientId, date, input);

    var surveyAnswers = surveyAnswersRepository.findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date).map(existingSurveyAnswers -> {
        return surveysServiceMapper.update(existingSurveyAnswers, input);
    })
    .map(surveyAnswersRepository::save)
    .orElseThrow();
    return surveyAnswers;


}

    @Transactional

    public Optional<SurveyAnswers> updateSurveyAnswers(Long surveyId, Long patientId, LocalDate date, java.util.Map input)
 {
    log.debug("Request updateSurveyAnswers: {} {} {} {}", surveyId, patientId, date, input);

    var surveyAnswers = surveyAnswersRepository.findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date).map(existingSurveyAnswers -> {
    return surveysServiceMapper.update(existingSurveyAnswers, input);
    })
    .map(surveyAnswersRepository::save)
    ;
    return surveyAnswers;


}


    public Optional<SurveyAnswers> getSurveyAnswers(Long surveyId, Long patientId, LocalDate date)
 {
    log.debug("Request getSurveyAnswers: {} {} {}", surveyId, patientId, date);
    var surveyAnswers = surveyAnswersRepository.findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date);
    return surveyAnswers;


}



}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/impl/SurveysServiceImpl.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service.impl

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.impl.mappers.*
import java.math.*
import java.time.*
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/** Service Implementation for managing [SurveyAnswers]. */
@Service
@Transactional(readOnly = true)
open class SurveysServiceImpl(private val surveyAnswersRepository: SurveyAnswersRepository) : SurveysService {

    private val log: Logger = LoggerFactory.getLogger(javaClass)

    private val surveysServiceMapper: SurveysServiceMapper = SurveysServiceMapper.INSTANCE

    override fun getSurveyAndQuestionsForPatient(name: String, patientId: Long, lang: String?): SurveyAndQuestions {
        log.debug("Request getSurveyAndQuestionsForPatient: {} {} {}", name, patientId, lang)

        val surveyAnswers = surveysServiceMapper.update(SurveyAnswers(), name, patientId, lang)
        // TODO: implement this method
        return surveysServiceMapper.asSurveyAndQuestions(surveyAnswers)
    }

    override fun answerSurvey(surveyId: Long, patientId: Long, date: LocalDate, input: SurveyAnswers): SurveyAnswers {
        log.debug("Request answerSurvey: {} {} {} {}", surveyId, patientId, date, input)

        return surveyAnswersRepository
            .findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date)
            ?.let { existingSurveyAnswers -> surveysServiceMapper.update(existingSurveyAnswers, input) }
            ?.let { surveyAnswersRepository.save(it) }
            ?: throw NoSuchElementException(
                "SurveyAnswers not found with id: surveyId=$surveyId, patientId=$patientId, date=$date"
            )
    }

    @Transactional
    override fun updateSurveyAnswers(
        surveyId: Long,
        patientId: Long,
        date: LocalDate,
        input: Map<String, Any?>,
    ): SurveyAnswers? {
        log.debug("Request updateSurveyAnswers: {} {} {} {}", surveyId, patientId, date, input)

        return surveyAnswersRepository
            .findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date)
            ?.let { existingSurveyAnswers -> surveysServiceMapper.update(existingSurveyAnswers, input) }
            ?.let { surveyAnswersRepository.save(it) }
    }

    override fun getSurveyAnswers(surveyId: Long, patientId: Long, date: LocalDate): SurveyAnswers? {
        log.debug("Request getSurveyAnswers: {} {} {}", surveyId, patientId, date)
        val surveyAnswers = surveyAnswersRepository.findBySurveyIdAndPatientIdAndDate(surveyId, patientId, date)
        return surveyAnswers
    }
}
